core/node/rules/can_add_event.go

```diff
@@ -1280,6 +1307,54 @@ func (ru *aeUnpinRules) validUnpin() (bool, error) {
 	return false, RiverError(Err_INVALID_ARGUMENT, "message is not pinned")
 }
 
+type HasChannelIdBytes interface {
+	ChannelIdBytes() ([]byte, error)
+}
+
+type autojoinWrapper struct {
+	update *SpacePayload_UpdateChannelAutojoin
+}
```

Comment:
I really like this! can we follow the convention and make a aeSpaceUpdateChannelRules struct at the top of this file?

Can we move the HasChannelIdBytes interface to the top of the file?

Why is HasChannelIdBytes public?

Can we follow convention for hideUserJoinLeaveEventsWrapper as well?